src/handwritten/_grpc_time.py

```diff
@@ -13,10 +16,32 @@
 _YS_PER_NS = 10**15
 _YS_PER_FS = 10**9
 
+def convert_to_desired_timezone(expected_time_utc, tzinfo):
+    current_time_utc = ht_datetime.now(timezone.utc)
+    desired_timezone_offset = current_time_utc.astimezone(tz=tzinfo).utcoffset()
+    desired_expected_time = expected_time_utc + desired_timezone_offset
+    new_datetime = ht_datetime(
+        desired_expected_time.year,
+        desired_expected_time.month,
+        desired_expected_time.day,
+        desired_expected_time.hour,
+        desired_expected_time.minute,
+        desired_expected_time.second,
+        desired_expected_time.microsecond,
+        desired_expected_time.femtosecond,
+        desired_expected_time.yoctosecond,
+        tzinfo = tzinfo)
+    return new_datetime
 
-def convert_time_to_timestamp(dt: Union[std_datetime, ht_datetime], ts: GrpcTimestamp) -> None:
-    utc_dt = dt.astimezone(tz=timezone.utc)
-    seconds = int(utc_dt.timestamp())
+
+def convert_time_to_timestamp(dt: Union[std_datetime, ht_datetime], ts: Optional[GrpcTimestamp] = None) -> GrpcTimestamp:
+    epoch_1970 = ht_datetime(1970, 1, 1, tzinfo=timezone.utc)
+    if dt < epoch_1970:
+        seconds_since_1970 = -int(abs(dt - epoch_1970).total_seconds())
+    else:
+        seconds_since_1970 = int(abs(dt - epoch_1970).total_seconds())
```

Comment:
Are you sure the conditional is necessary? The sign appears to be correct without it.

```
>>> (datetime.datetime(1969, 1, 1) - datetime.datetime(1970, 1, 1)).total_seconds()
-31536000.0
>>> (datetime.datetime(1971, 1, 1) - datetime.datetime(1970, 1, 1)).total_seconds()
31536000.0
```